src/commands/login-new.js

```diff
@@ -0,0 +1,163 @@
+
+// Not really checked right now, but it might come useful if we ever need to do some breaking changes
+const API_VERSION = 'v1';
+
+let tokenPrompt;
+let promptUi;
+
+const tryToLogin = async (token, fromPrompt) => {
+    const isUserLogged = await getLoggedClient(token, API_BASE_URL);
+    const userInfo = getLocalUserInfo();
+    if (isUserLogged) {
+        outputs.success(`You are logged in to Apify as ${userInfo.username || userInfo.id}!`);
+        if (fromPrompt) process.exit();
+        promptUi.close();
+    } else {
+        outputs.error('Login to Apify failed, the provided API token is not valid.');
+    }
+    return isUserLogged;
+};
+
+class LoginNewCommand extends ApifyCommand {
+    async run() {
+        outputs.warning('This command is still experimental and might break at any time. Use at your own risk.\n');
+
+        const { flags } = this.parse(LoginNewCommand);
+        let { token } = flags;
+        if (!token) {
+            let server;
+            const app = express();
+
+            // To send requests from browser to localhost, CORS has to be configured properly
+            app.use(cors({
+                origin: CONSOLE_URL_ORIGIN,
+                allowedHeaders: ['Content-Type', 'Authorization'],
+            }));
+
+            // Turn off keepalive, otherwise closing the server when command is finished is lagging
+            app.use((req, res, next) => {
+                res.set('Connection', 'close');
+                next();
+            });
+
+            app.use(express.json());
+
+            // Basic authorization via a random token, which is passed to the Apify Console,
+            // and that sends it back via the `token` query param, or `Authorization` header
+            const authToken = cryptoRandomObjectId();
+            app.use((req, res, next) => {
+                let { token: serverToken } = req.query;
+                if (!serverToken) {
+                    const authorizationHeader = req.get('Authorization');
+                    if (authorizationHeader) {
+                        const [schema, tokenFromHeader, ...extra] = authorizationHeader.trim().split(/\s+/);
+                        if (schema.toLowerCase() === 'bearer' && tokenFromHeader && extra.length === 0) {
+                            serverToken = tokenFromHeader;
+                        }
+                    }
+                }
+
+                if (serverToken !== authToken) {
+                    res.status(401);
+                    res.send('Authorization failed');
+                } else {
+                    next();
+                }
+            });
+
+            const apiRouter = express.Router();
+            app.use(`/api/${API_VERSION}`, apiRouter);
+
+            apiRouter.post('/login-token', async (req, res) => {
+                try {
+                    if (req.body.apiToken) {
+                        outputs.info('cancelled\nGot token from console...');
```

Comment:
Maybe this would format better:
```suggestion
                        console.log('canceled');
                        outputs.info('Got token from console...');
```